src/utils/calculateBasicAuthUsernamePasswordPair.js

```diff
@@ -0,0 +1,32 @@
+import { oauth2 } from './oauth2'
+
+export function calculateBasicAuthUsernamePasswordPair ({
+  username,
+  password,
+  token,
+  oauth2format
+}) {
+  const key = [!!username, !!password, !!token, !!oauth2format]
+    .map(Number)
+    .join('')
+  // See helpful truth table in documentation
+  // prettier-ignore
+  switch (key) {
+    case '0000': return null
+    case '1000': throw new Error(`Missing password or token`)
+    case '0100': throw new Error(`Missing username`)
+    case '1100': return { username, password }
+    case '0010': return { username: token, password: '' }
```

Comment:
This deserves at least a comment. :-)

Reply:
That said I am not opposed to this shortcut, although I do find myself wishing we could only limit it to GitHub smarter - but I don't see how we could do it. I figured host check in the URL would work, but that would fail with enterprise GitHub deployments. So I think let's leave this in and see if people get confused by this or not.